scripts/gen_hashed_bin.py

```diff
@@ -35,14 +35,25 @@
 
 arch_id = {'arm32': 0, 'arm64': 1}
 
-def write_header(outf, init_size, args, paged_size):
+def write_header_v1(outf, init_size, args, paged_size):
 	magic = 0x4554504f # 'OPTE'
 	version = 1;
 	outf.write(struct.pack('<IBBHIIIII', \
 		magic, version, arch_id[args.arch], args.flags, init_size, \
 		args.init_load_addr_hi, args.init_load_addr_lo, \
 		args.init_mem_usage, paged_size))
-	
+
+def write_header_v2(outf, init_size, args, paged_size):
+	magic = 0x4554504f # 'OPTE'
+	version = 2
+	nb_images = 1 if paged_size == 0 else 2
+	outf.write(struct.pack('<IBBHI', \
+		magic, version, arch_id[args.arch], args.flags, nb_images))
+	outf.write(struct.pack('<IIII', \
+		args.init_load_addr_hi, args.init_load_addr_lo, 0, init_size))
```

Comment:
I would rather replace magic value 0 (and 1 below) with explicit IDs, whatever they being 0 or not.